content-assets/wcag-act-rules/filter-scripts.js

```diff
@@ -0,0 +1,422 @@
+        // Test Type Logic:
+        const ruleTypes = rule.dataset.testTypes ? rule.dataset.testTypes.split(' ') : [];
+        const shouldShowByTestType = (
+          (ruleTypes.includes('automated') && filters.showAutomated) ||
+          (ruleTypes.includes('semiauto') && filters.showSemiauto) ||
+          (ruleTypes.includes('manual') && filters.showManual) ||
+          (ruleTypes.includes('linter') && filters.showLinter) ||
+          (ruleTypes.includes('unimplemented') && filters.showUnimplemented) ||
+          (ruleTypes.length === 0 && filters.showUnimplemented)
+        );
+        let shouldShow = shouldShowByStatus && shouldShowByTestType;
+        toggleVisibility(rule, shouldShow);
+        if (shouldShow) visibleRuleCount++;
+      });
+      return visibleRuleCount;
+    }
+
+    /**
+     * Filter rules within a criterion based on their status and test type
+     * @returns {boolean} - True if any rules are visible after filtering, false otherwise.
+     */
+    function filterRules(criterion, filters) {
+      const ruleList = criterion.querySelector('ul.act-rule-list');
+      const visibleRuleCount = filterRuleList(ruleList, filters);
+      // Handle "No ACT Rules available" message and rule list
+      const noRulesMessage = criterion.querySelector('p > em');
+      const filteredMessage = criterion.querySelector('.filtered-rules-message'); 
+      const relatedRulesPara = ruleList ? ruleList.previousElementSibling?.previousElementSibling : null; // Safer selection
+      if (ruleList && relatedRulesPara && filteredMessage) {
+        if (visibleRuleCount > 0) {
+          // Some rules are visible
+          toggleVisibility(ruleList, true);
+          toggleVisibility(relatedRulesPara, true);
+          toggleVisibility(filteredMessage, false);
+        } else {
+          // No rules exist for this criterion (visibleRuleCount === 0)
+          toggleVisibility(ruleList, false);
+          toggleVisibility(relatedRulesPara, false);
+          toggleVisibility(filteredMessage, false);
+          if (noRulesMessage && noRulesMessage.parentElement) {
+            toggleVisibility(noRulesMessage.parentElement, true);
+          }
+        }
+      } else if (noRulesMessage && noRulesMessage.parentElement) {
+        // Original case: No rule list exists, ensure 'No rules' message is visible
+        toggleVisibility(noRulesMessage.parentElement, true);
+      }
+      return visibleRuleCount > 0; // Return whether rules are visible
+    }
+    
+    /**
+     * Updates the visibility of guideline sections based on whether they have any visible criteria.
+     * Assumes each <h2 class="guideline-section"> is followed by its .sc-item elements.
+     * Hides the guideline if none of its .sc-item children are visible.
+     */
+    function updateSectionVisibility() {
+      const guidelines = document.querySelectorAll('h2.guideline-section, h2[data-aria-section]');
```

Comment:
We should update this because, as said in other comments, I don't think the proposed heading structure is right.